src/ImageSharp/ImageFrame{TPixel}.cs

```diff
@@ -148,13 +148,22 @@ internal ImageFrame(Configuration configuration, ImageFrame<TPixel> source)
         /// <param name="x">The x-coordinate of the pixel. Must be greater than or equal to zero and less than the width of the image.</param>
         /// <param name="y">The y-coordinate of the pixel. Must be greater than or equal to zero and less than the height of the image.</param>
         /// <returns>The <see typeparam="TPixel"/> at the specified position.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the provided (x,y) coordinates are outside the image boundary.</exception>
         public TPixel this[int x, int y]
         {
-            [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            get => this.PixelBuffer[x, y];
+            [MethodImpl(InliningOptions.ShortMethod)]
+            get
+            {
+                this.VerifyCoords(x, y);
```

Comment:
@James-Jackson-South I added proper bounds checking here. To compensate the performance penalty, I tried to make the underlying `GetElementUnsafe()` implementation as fast as possible, but I expect the aggregate performance change to be still negative. Still don't think we should omit proper input validation on a method with this level of visibility.

I would advise everyone to use `.GetRowSpan(y)` anyways, `[x,y]` is slow by design, whatever we do.

Reply:
`[x,y]` is there for convenience in the most part, though it's used a lot in drawing and in algorithms where we are not operating on the current row span. Error diffusion for example. If you think perf is not affected too badly then I'm up for it.

Reply:
`[x,y]` should be used on `Buffer2D<T>` in processor code, which has been optimized now as much as we can.